tests/integration/olp-cpp-sdk-dataservice-read/VolatileLayerClientTest.cpp

```diff
@@ -0,0 +1,835 @@
+  olp::client::HRN hrn(GetTestCatalog());
+
+  EXPECT_CALL(*network_mock_, Send(IsGetRequest(URL_PARTITIONS), _, _, _, _))
+      .Times(0);
+
+  olp::dataservice::read::VolatileLayerClient client(hrn, "testlayer_volatile",
+                                                     settings_);
+
+  auto request = olp::dataservice::read::PartitionsRequest().WithFetchOption(
+      FetchOptions::CacheOnly);
+  VolatileLayerClient::PartitionsResponse partitions_response;
+  olp::client::Condition condition;
+  client.GetPartitions(request,
+                       [&](VolatileLayerClient::PartitionsResponse response) {
+                         partitions_response = std::move(response);
+                         condition.Notify();
+                       });
+
+  ASSERT_TRUE(condition.Wait(kTimeout));
+
+  ASSERT_FALSE(partitions_response.IsSuccessful())
+      << ApiErrorToString(partitions_response.GetError());
+}
+
+TEST_F(VolatileLayerClientReadTest, GetPartitionsOnlineOnly) {
+  olp::client::HRN hrn(GetTestCatalog());
+
+  {
+    testing::InSequence s;
+
+    EXPECT_CALL(*network_mock_, Send(IsGetRequest(URL_CONFIG), _, _, _, _))
+        .Times(1);
+
+    EXPECT_CALL(*network_mock_, Send(IsGetRequest(URL_CONFIG), _, _, _, _))
+        .WillOnce(
+            ReturnHttpResponse(olp::http::NetworkResponse().WithStatus(429),
+                               "Server busy at the moment."));
+  }
+
+  olp::dataservice::read::VolatileLayerClient client(hrn, "testlayer",
+                                                     settings_);
+
+  auto request = olp::dataservice::read::PartitionsRequest().WithFetchOption(
+      FetchOptions::OnlineOnly);
+  {
+    VolatileLayerClient::PartitionsResponse partitions_response;
+    olp::client::Condition condition;
+    client.GetPartitions(request,
+                         [&](VolatileLayerClient::PartitionsResponse response) {
+                           partitions_response = std::move(response);
+                           condition.Notify();
+                         });
+
+    ASSERT_TRUE(condition.Wait(kTimeout));
+
+    ASSERT_TRUE(partitions_response.IsSuccessful())
+        << ApiErrorToString(partitions_response.GetError());
+    ASSERT_EQ(4u, partitions_response.GetResult().GetPartitions().size());
+  }
+
+  {
+    VolatileLayerClient::PartitionsResponse partitions_response;
+    olp::client::Condition condition;
+    client.GetPartitions(request,
+                         [&](VolatileLayerClient::PartitionsResponse response) {
+                           partitions_response = std::move(response);
+                           condition.Notify();
+                         });
+
+    ASSERT_TRUE(condition.Wait(kTimeout));
+    // Should fail despite valid cache entry
+    ASSERT_FALSE(partitions_response.IsSuccessful())
+        << ApiErrorToString(partitions_response.GetError());
+  }
+}
+
+TEST_F(VolatileLayerClientReadTest, DISABLED_GetPartitionsCacheWithUpdate) {
```

Comment:
```suggestion
TEST_F(VolatileLayerClientReadTest, DISABLED_GetPartitionsCacheWithUpdate) {
```
Have you checked what was the reason that it is disabled?